subsys/bluetooth/host/direction.c

```diff
@@ -131,3 +229,63 @@ static int hci_df_set_conn_cte_tx_param(struct bt_conn *conn, uint8_t cte_types,
 
 	return err;
 }
+
+/* @brief Function initializes Direction Finding in Host
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+int le_df_init(void)
+{
+	uint8_t max_switch_pattern_len;
+	uint8_t switch_sample_rates;
+	uint8_t max_cte_len;
+	uint8_t num_ant;
+	int err;
+
+	err = hci_df_read_ant_info(&switch_sample_rates, &num_ant,
+			     &max_switch_pattern_len, &max_cte_len);
+	if (err) {
+		return err;
+	}
+
+	df_ant_info.max_switch_pattern_len = max_switch_pattern_len;
+	df_ant_info.switch_sample_rates = switch_sample_rates;
+	df_ant_info.max_cte_len = max_cte_len;
+	df_ant_info.num_ant = num_ant;
+
+	BT_INFO("DF initialized.");
+	return 0;
+}
+
+int bt_le_df_set_adv_cte_tx_params(struct bt_le_ext_adv *adv,
+				const struct bt_le_df_adv_cte_tx_params *params)
+{
+	__ASSERT_NO_MSG(adv);
+	__ASSERT_NO_MSG(params);
+
+	int err;
+
+	if (!BT_FEAT_LE_CONNECTIONLESS_CTE_TX(bt_dev.le.features)) {
+		return -ENOTSUP;
+	}
+
+	/* Check if BT_ADV_PARAMS_SET is set, because it implies the set
+	 * has already been created.
+	 */
+	if (!atomic_test_bit(adv->flags, BT_ADV_PARAMS_SET)) {
+		return -EINVAL;
+	}
+
+	if (atomic_test_bit(adv->flags, BT_PER_ADV_CTE_ENABLED)) {
+		return -EALREADY;
+	}
+
+	err = hci_df_set_cl_cte_tx_params(adv, params);
+	if (err) {
+		return err;
+	}
+
+	atomic_set_bit(adv->flags, BT_PER_ADV_CTE_PARAMS_SET);
```

Comment:
This is never read, do you need it?

If it was supposed to be CTE_ENABLED I would suggest to set it using state_set mechanism so that it is set during the handling of the command complete event

Reply:
It is needed for function that will handle CTE enable command.

Reply:
If this will be used in a future addition then it should be moved to that commit so that the commits are self-contained.

Reply:
@joerchan Isn't it self-contained? I assume the commit shoud provide complete implementation of some functionality. In this case it is implementation of `bt_le_df_set_adv_cte_tx_params`. This is part of implementation of the function.
If I provide it as part of implementation of `bt_df_set_adv_cte_tx_enabled` then one may ask: "why implementation of enable function includes changes in other functions?". 
I thought the goal is to provide complete implementation if that is possible. 

Am I wrong here?

Reply:
Because they don't do anything at the moment. You should add the flags when you use them. 
Self-contained as a feature, not a function.

Reply:
Well, I've added them now, because for me it was part of complete feature - to set up CTE parameters (sorry for being unclear). I can remove it if you whish that and add it in following PR.
I just want to get feeling what is expected to be delivered in PRs:
- complete feature as it is required by protocol specification, this means for me add these flags here because they are part of specification and can be added.
-  complete feature in the meaning of code that is actually in use? So not all things from protocol specification are be implemented but no unused code is present.

Reply:
@ppryga I'm mostly referring to this from the contributing guidelines:
```
Each commit must address a single identifiable issue and must be logically self-contained. Unrelated changes should be submitted as separate commits.
```

The reason why I wanted them to be grouped together is that when you look at just the single commit it looks really strange, it looks like something has been forgotten and a possible bug
And when the follow-up would have been commit in a PR later there could be hundreds of commits in between.
Someone looking through the history, either debugging or curious might, might be confused and think the same as I did.

I think the easiest would have been to add `bt_df_set_adv_cte_tx_enabled` in the same commit as you added `bt_df_set_adv_cte_tx_param`. That way you could  look at the all together.

Sorry if I'm making too big a deal of out it.